aea/protocols/generator.py

```diff
@@ -19,13 +19,14 @@
 """This module contains the protocol generator."""
 
 import os
+import re
 from os import path
 from pathlib import Path
 from typing import Dict, List
 
 from aea.configurations.base import ProtocolSpecification
 
-DEFAULT_TYPES = ["int", "float", "bool", "str", "bytes", "list", "dict", "tuple", "set"]
+CUSTOM_TYPE_PATTERN = "[A-Z_][a-zA-Z0-9]*"
```

Comment:
I don't know why we you went for this regex. Care to elaborate?

Reply:
Also, see comment further below.

Reply:
Originally I found this https://stackoverflow.com/questions/10120295/valid-characters-in-a-python-class-name (see second answer for regex) and made some modifications. 

This format is something we can agree together. 

The format of class names from the official Python language reference is here: https://docs.python.org/3/reference/lexical_analysis.html#identifiers

But as you see that is very broad. 

A couple of options:

- [A-Z_][a-zA-Z0-9_]*
- [A-Z][a-zA-Z0-9_]*[a-zA-Z0-9]

After a bit of thinking, I am leaning towards the second regex above.

When choosing this, we should try to be language independent, so not necessarily something very Pythonic.

Reply:
[A-Z][a-zA-Z0-9]*

Reply:
Done